src/Equinox.MessageDb/MessageDb.fs

```diff
@@ -322,63 +350,94 @@ type MessageDbContext(connection : MessageDbConnection, batchOptions : BatchOpti
 
     member _.TrySync(log, streamName, token, encodedEvents : IEventData<EventBody> array): Async<GatewaySyncResult> = async {
         let streamVersion = Token.streamVersion token
-        match! Write.writeEvents log connection.WriteRetryPolicy connection.Writer streamName streamVersion encodedEvents with
+        match! Write.writeEvents log connection.WriteRetryPolicy connection.Writer streamName (Some streamVersion) encodedEvents with
         | MdbSyncResult.ConflictUnknown ->
             return GatewaySyncResult.ConflictUnknown
         | MdbSyncResult.Written version' ->
             let token = Token.create version'
             return GatewaySyncResult.Written token }
 
 [<NoComparison; NoEquality; RequireQualifiedAccess>]
-type AccessStrategy =
+type AccessStrategy<'event, 'state> =
     /// Load only the single most recent event defined in in a stream and trust that it'll be decoded and
     /// doing a <c>fold</c> from any such event will yield a correct and complete state
     /// In other words, the <c>fold</c> function should not need to consider either the preceding <c>'state</c> or <c>'event</c>s.
     | LatestKnownEvent
+    /// Ensures a snapshot/compaction event from which the state can be reconstituted upon decoding is always present
+    /// (embedded in a special snapshot stream as an event), generated every <c>batchSize</c> events using the supplied
+    /// <c>toSnapshot</c> function.
+    /// When loading the stream it'll first fetch the snapshot.
+    | RollingSnapshots of originType: string * toSnapshot : ('state -> 'event)
```

Comment:
needs a different name, maybe `SnapshotStream`

Reply:
note Rolling Snapshots, per the readme and/or DOC.md is referencing the semi-canonical definition of it in the esdb docs, which explicitly mentions embedding them in the main stream 

https://eventstore.org/docs/event-sourcing-basics/rolling-snapshots/index.html

All I could find was https://web.archive.org/web/20180626025148/https://eventstore.org/docs/event-sourcing-basics/rolling-snapshots/index.html

Perhaps this calls for the readme to have a section explaining snapshots can be:
- Embedded (SSS, ESDB) - needs to have a big warning about mixing state into event streams not being a bright idea in general
- Adjacent/Sibling - explains this is only available for mdb atm
- External (customisable external call - as that's not implemented, no need to mention)

Perhaps linking to this and using its terms might make sense ? https://www.eventstore.com/blog/snapshotting-strategies

Given we are going to V4 and Rolling Snapshots no longer seems to mean anything canonical anywhere, it might make sense to rename the DU cases for SSS and ESDB now to make it self-explanatory even without having read the relevant readme section